clash-lib/src/Clash/Rewrite/Util.hs

```diff
@@ -201,9 +202,17 @@ apply = \s rewrite ctx expr0 -> do
 
   if lvl == DebugNone
     then return expr2
-    else applyDebug lvl dbgTranss fromLimit s expr0 hasChanged expr2
+    else do
+      Monad.when (lvl >= DebugCount && hasChanged) $ updateTransformCounters s
+      applyDebug lvl dbgTranss fromLimit s expr0 hasChanged expr2
 {-# INLINE apply #-}
 
+updateTransformCounters
+  :: String -- ^ Name of the transformation
+  -> RewriteMonad extra ()
+updateTransformCounters name =
+  transformCounters  %= (MapStrict.insertWith mappend name (Monoid.Sum 1))
```

Comment:
Any reason to not just use an `Word` and:

```haskell
transformCounters  %= MapStrict.insertWith (\_ -> succ) name 1
```

?

Reply:
I started with lenses to go into the Map, using the [`<>=` combinator](https://ekmett.github.io/lens/Control-Lens-Setter.html#v:-60--62--61-), which needed the Monoid.
But later decided against it because the Map lenses are lazy and I think I want the Map to be strict.

Reply:
Leaving this as a `Sum Word` unless there are any strong objections. It shouldn't cost anything extra because it's a `newtype` anyway and GHC should be able to optimize things like `(<>) = coerce (+)` down to just addition anyway

Reply:
I think we should change it. Abstractions shouldn't be used if not necessary, it only adds cognitive load for the reader.

Reply:
Sure, whatever. Changed it to `HashMap Text Word`